Vienna/Sources/Models/FolderImageCache.m

```diff
@@ -36,110 +36,127 @@ + (FolderImageCache *)defaultCache
+    for (NSString *filenameWithExtension in filenames) {
+        if (![filenameWithExtension.pathExtension isEqualToString:@"tiff"]) {
+            continue;
         }
-        if (imageData != nil) {
-            [[NSFileManager defaultManager] createFileAtPath:fullFilePath contents:imageData attributes:nil];
+
+        NSString *filePath =
+            [cacheDirectoryPath stringByAppendingPathComponent:filenameWithExtension];
+        NSImage *image = [[NSImage alloc] initWithContentsOfFile:filePath];
+        if (image && image.isValid) {
+            NSString *filename = filePath
+                                     .lastPathComponent
+                                     .stringByDeletingPathExtension
+                                     .vna_convertStringToValidPath;
+            _folderImages[filename] = image;
         }
     }
+    return _folderImages;
 }
 
-/* retrieveImage
- * Retrieve the image for the specified URL from the cache.
- */
--(NSImage *)retrieveImage:(NSString *)baseURL
+- (BOOL)cacheImageData:(NSData *)imageData
+              filename:(NSString *)filename
 {
-    [self initFolderImagesArray];
-    return folderImagesArray[baseURL];
-}
+    NSImage *image = [[NSImage alloc] initWithData:imageData];
+    if (!image) {
+        return NO;
+    }
 
-/* initFolderImagesArray
- * Load the existing list of folder images from the designated folder image cache. We
- * do this only once and we do it as quickly as possible. When we're done, the folderImagesArray
- * will be filled with image representations for each valid image file we find in the cache.
- */
--(void)initFolderImagesArray
-{
-    if (!initializedFolderImagesArray) {
-        NSFileManager * fileManager = [NSFileManager defaultManager];
-        NSArray * listOfFiles;
-        BOOL isDir;
-        
-        // Get and cache the path to the folder. This is the best time to make sure it
-        // exists. The penalty for it not existing AND us being unable to create it is that
-        // we don't cache folder icons in this session.
-        NSURL *appSupportURL = fileManager.vna_applicationSupportDirectory;
-        NSURL *imagesURL = [appSupportURL URLByAppendingPathComponent:@"Images"
-                                                          isDirectory:YES];
-        imagesCacheFolder = imagesURL.path;
-        if (![fileManager fileExistsAtPath:imagesCacheFolder isDirectory:&isDir]) {
-            if (![fileManager createDirectoryAtPath:imagesCacheFolder withIntermediateDirectories:YES attributes:nil error:nil]) {
-                NSLog(@"Cannot create image cache at %@. Will not cache folder images in this session.", imagesCacheFolder);
-                imagesCacheFolder = nil;
-            }
-            initializedFolderImagesArray = YES;
-            return;
+    // macOS 15 does not handle images with indexed pixel samples correctly.
+    // Those images have to be converted.
```

Comment:
The comment would be clearer if it explicitly stated that we always store in RGB color space